orm/versioning_bucket_test.go

```diff
@@ -0,0 +1,51 @@
+package orm
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+)
+
+func TestGetLatestVersion(t *testing.T) {
+	bucketImpl := NewBucket("any", NewSimpleObj(nil, &VersionedIDRef{}))
+	idGenBucket := WithSeqIDGenerator(bucketImpl, "id")
+	versionedBucket := WithVersioning(idGenBucket)
+	db := store.MemStore()
+	// when
+	obj, err := versionedBucket.Create(db, &VersionedIDRef{ID: []byte("anyValue")})
+	if err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	var vID VersionedIDRef
+	if err := vID.Unmarshal(obj.Key()); err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	// test some iterations
+	for i := 0; i < 100; i++ {
```

Comment:
Well, you really need to worry about the second byte in varint encoding... So, over 127. Go to at least 200 for the ordering issue to appear. Ideally over 256 so little vs big-endian issues also show up